lib/native_enum/activerecord_enum_post42.rb

```diff
@@ -3,29 +3,31 @@ module ConnectionAdapters
     if defined?(AbstractMysqlAdapter)
       class AbstractMysqlAdapter
         protected
-        def initialize_type_map_with_enum(m = type_map)
-          initialize_without_enum(m)
-          register_enum_type(m, %r(^enum)i)
-          register_set_type(m, %r(^set)i)
-        end
+        self << class
+          def initialize_type_map_with_enum(m = type_map)
+            initialize_without_enum(m)
+            register_enum_type(m, %r(^enum)i)
+            register_set_type(m, %r(^set)i)
+          end
 
-        alias_method :initialize_without_enum, :initialize_type_map
```

Comment:
`initialize_type_map` is a class method in Rails 7.x, so this no longer works after upgrading Rails.